Dockerfile

```diff
@@ -1,25 +1,46 @@
-FROM python:3.9-alpine
+FROM python:3.9-alpine as buildimage
 
-VOLUME [ "/config"]
+COPY . /tmp/app_install
 
-# Install required dependencies
-RUN apk add --no-cache \
-# Support for Timezones
-    tzdata \
+RUN set -eux;\
+	apk add --no-cache \
 # ujson won't compile without these libs
-    g++
+		g++; \
+# wheel all packages for habapp
+	cd /tmp/app_install; \
+	pip wheel --wheel-dir=/root/wheels --use-feature=in-tree-build .
 
-# Always use latest versions
-RUN mkdir -p /usr/src/app
-WORKDIR /usr/src/app
-COPY . .
+FROM python:3.9-alpine
```

Comment:
Why is this used again and not buildimage?

Reply:
This is the beginning of the final container. Everything added inside the buildimage part wan't be available to the final container unless you copy it with COPY --from=buildimage. For example in the final container image there will be no /tmp/app_install directory and no g++ and it's dependencies is installed. This way the size of the final container image is reduced a lot.